hardware_interface/include/hardware_interface/component_interfaces/system_interface.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__COMPONENT_INTERFACES__SYSTEM_INTERFACE_HPP_
+#define HARDWARE_INTERFACE__COMPONENT_INTERFACES__SYSTEM_INTERFACE_HPP_
+
+#include <string>
+#include <vector>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+class SystemInterface
+{
+public:
+  HARDWARE_INTERFACE_PUBLIC
+  SystemInterface() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  ~SystemInterface() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  return_type configure(const SystemInfo & system_info) = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  std::vector<std::string> get_interface_names() const = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  return_type start() = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  return_type stop() = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  bool is_started() const = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  return_type read(std::vector<double> & data) = 0;
+
+  return_type
+  virtual
+  write(const std::vector<double> & data) = 0;
```

Comment:
this vector data comes from the resource manager? Or which entity is allocating the memory for the data?